source/val/validation_state.h

```diff
@@ -358,6 +358,11 @@ class ValidationState_t {
       uint32_t id, uint32_t* num_rows, uint32_t* num_cols,
       uint32_t* column_type, uint32_t* component_type) const;
 
+  // Collects struct member types into |member_types|.
+  // Returns false if parsing fails (wrong syntaxis or not struct type).
```

Comment:
"syntaxis"?

Reply:
Also returns false if the struct has no members.

Also, it's probably ok that the vector is not updated in the failure case.  Probably better to say that in the method comment as well.